ibtk/include/ibtk/HierarchyAveragedDataManager.h

```diff
@@ -0,0 +1,293 @@
+ *         hier_math_ops.resetLevels(0, patch_hierarchy->getFinestLevelNumber());
+ *         const int wgt_sc_idx = hier_math_ops.getSideWeightPatchDescriptorIndex();
+ *         bool at_steady_state = hier_avg_data_manager.updateTimeAveragedSnapshot(
+ *                  u_idx, t, patch_hierarchy, "CONSERVATIVE_LINEAR_REFINE", wgt_sc_idx, 1.0e-8);
+ *         if (at_steady_state)
+ *             pout << "Determined steady state!\n";
+ *     }
+ * }
+ * \endcode
+ *
+ */
+class HierarchyAveragedDataManager
+{
+public:
+    /*!
+     * The constructor for class HierarchyAveragedDataManager sets some default values and determines data centering.
+     * The expected period and number of snapshots must be set in the input database. This class assumes the snapshots
+     * will be taken at equidistant points along the periodic interval.
+     *
+     * The input database is searched for the following keys:
+     *  - 't_start' : Double that represents the beginning of the period
+     *  - 't_end' : Double that represents the end of the period. For cases where the period is zero, t_end should be
+     * set to t_start.
+     *  - 'threshold' : Double to determine whether a given time point is at a periodic steady state.
+     *  - 'num_snapshots' : Integer to determine the number of snapshots taken, equally sampled between t_start and
+     * t_end.
+     *  - 'enable_logging' : Bool used to determine whether to print convergence information to the log file.
+     *  - 'output_data' : Bool used to determine whether to write visualization files for the mean and deviations.
+     *  - 'dir_dump_name' : String used to determine which folder to write visualization files to.
+     *  - 'refine_type' : String to determine which refine operator will be used by default.
+     */
+    HierarchyAveragedDataManager(std::string object_name,
+                                 SAMRAI::tbox::Pointer<SAMRAI::hier::Variable<NDIM> > var,
+                                 SAMRAI::tbox::Pointer<SAMRAI::tbox::Database> input_db,
+                                 SAMRAI::tbox::Pointer<SAMRAI::hier::PatchHierarchy<NDIM> > hierarchy,
+                                 SAMRAI::tbox::Pointer<SAMRAI::hier::GridGeometry<NDIM> > grid_geom,
+                                 bool register_for_restart = true);
+
+    /*!
+     * The constructor for class HierarchyAveragedDataManager sets some default values and determines data centering. In
+     * this constructor, the times at which snapshots are taken are set by arguments.
+     *
+     * The input database is searched for the following keys:
+     *  - 'enable_logging' : Bool used to determine whether to print convergence information to the log file.
+     *  - 'output_data' : Bool used to determine whether to write visualization files for the mean and deviations.
+     *  - 'dir_dump_name' : String used to determine which folder to write visualization files to. Defaults to "".
```

Comment:
```suggestion
     *  - 'dir_dump_name' : String used to determine which folder to write visualization files to. Defaults to "./".
```
What you have works but we can make it clearer.